llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -7482,12 +7482,11 @@ VPRecipeBuilder::tryToWidenMemory(Instruction *I, ArrayRef<VPValue *> Operands,
     VPSingleDefRecipe *VectorPtr;
     if (Reverse) {
       // When folding the tail, we may compute an address that we don't in the
-      // original scalar loop and it may not be inbounds. Drop Inbounds in that
-      // case.
+      // original scalar loop: drop the GEP no-wrap flags in this case.
```

Comment:
```suggestion
      // original scalar loop: drop the GEP no-wrap flags in this case. Otherwise preserve existing flags without no-unsigned-wrap, as we will emit negative indices.
```